doc/04-Upgrading.md

```diff
@@ -3,6 +3,35 @@
 Specific version upgrades are described below. Please note that version upgrades are incremental.
 If you are upgrading across multiple versions, make sure to follow the steps for each of them.
 
+## Upgrading to Icinga DB v1.1.2
+
+Please apply the `1.1.2.sql` upgrade script to your database.
+For package installations, you can find this file at `/usr/share/icingadb/schema/mysql/upgrades/` or
+`/usr/share/icingadb/schema/pgsql/upgrades/`, depending on your database type.
+
+As the daemon checks the schema version, the recommended way to perform the upgrade is to stop the daemon, apply the
+schema upgrade and then start the new daemon version. If you want to minimize downtime as much as possible, it is safe
+to apply this schema upgrade while the Icinga DB v1.1.1 daemon is still running and then restart the daemon with the
+new version. Please keep in mind that depending on the distribution, your package manager may automatically attempt to
+restart the daemon when upgrading the package.
+
+### Upgrading the state_history Table (optional)
+
+Icinga DB crashes if hosts/services reach check attempt 256. The `check_attempt` column of the `state_history` table
+is too small to fit values greater 255. ([#655](https://github.com/Icinga/icingadb/issues/655))
```

Comment:
> It's just that if you read that section top to bottom, with the information you were given up to that sentence, you may ask yourself "well, if it still crashes, what does this update even do?".

-- https://github.com/Icinga/icingadb/pull/656#discussion_r1549488023

Reply:
Your claim sounds better with _if you read **only the 1st 1/3 of** that section_. Yes. There's a crash. And the update doesn't fix it.\* But we have good reasons and wrote them down in the upgrading docs.

\* Actually it does, you "just" have to apply two files.

Reply:
> Your claim sounds better with _if you read **only the 1st 1/3 of** that section_.

Well yes, people tend to read text from top to bottom. And usually it's a feature if understanding a sentence doesn't depend on information only given later in the text. Otherwise, you end up with a text you have to read multiple times until you understand it.

Reply:
I've prepared a suggestion for how this can be written in 1298c304cc2adff7c8b13b2f7e732bcda8bbcc67. It's based on the text provided in this PR (hence the `Co-authored-by`) but restructured so that it first explains why the schema upgrade is split and then mentions the options including the "may crash" warning if the upgrade is skipped. Apart from that, I've tweaked the wording in a few places.

Reply:
Good in general, but for max_check_attempts=256 (which already works btw.) the highest possible check_attempt is 255: https://github.com/Icinga/icinga2/blob/9e31b8b5590c6d67b7dd538b2c884bd377a4e486/lib/icinga/checkable-check.cpp#L233-L237